blog/wdio-blog.md

```diff
@@ -0,0 +1,284 @@
+
+- Author: [Raja Panidepu](https://github.com/rajapanidepu).
+
+# Article:
+
+## Motivation: 
+I work on a Mobile App building team which employs React Native Framework. Similar to how we write JavaScript code to develop ReactNative components for Android or iOS platform, we wanted to write our UI tests once and run on both the platforms. The answer we found is WebDriverIO + Appium.
+
+WebDriverIO is a Node.js implementation of WebDriver Protocol. WebDriverIO is the most popular Web UI testing framework, which also supports Native Mobile UI testing.
+
+The next part of the article discusses setting up WebDriverIO of the following scenarios:
+1. UI tests running locally using Emulators
+2. Using Cloud-based service to run on Emulators
+3. Using Cloud-based service to run on Real Devices
+4. Timeouts at various levels
+
+## Part 1: Setup WDIO to run tests native UI tests locally.
+Let's install WebDriverIO globally and create a basic WebDriverIO config.
+
+```console
+$ npm install webdriverio --save-dev
+$ wdio config
+```
+ 
+![Screenshot showing wdio config](wdio-config.png)
+
+Follow the prompts to create base WDIO configuration as shown in the above picture.
+
+Note: For now, set the base URL to the default value: http://localhost, we will update that later. Once done, let WebDriverIO install all the needed packages.
+
+Finally, below WebDriverIO config would be saved in `wdio.conf` file:
+
+```js
+exports.config = {
+  specs: [
+    './test/specs/**/ *.js'
+  ],
+  exclude: [],
+  maxInstances: 10,
+  capabilities: [{
+    maxInstances: 5,
+    browserName: 'firefox'
+  }],
+  sync: true,
+  logLevel: 'verbose',
+  coloredLogs: true,
+  deprecationWarnings: true,
+  bail: 0,
+  screenshotPath: './errorShots/',
+  baseUrl: 'http://localhost',
+  waitforTimeout: 10000,
+  connectionRetryTimeout: 90000,
+  connectionRetryCount: 3,
+  framework: 'jasmine',
+  jasmineNodeOpts: {
+    defaultTimeoutInterval: 10000,
+    expectationResultHandler: function (passed, assertion) {
+    }
+  }
+};
+```
+
+The above config is not yet ready for running mobile UI tests. let start tweaking it.
+
+WebDriverIO supports multiple services, of which Appium is a test automation framework for use with native, hybrid and mobile web apps. [WebDriverIO's Appium service](http://webdriver.io/guide/services/appium.html) lets you automagically run Appium server in the background, which passes on UI test commands to the mobile simulator/emulator.
+
+Setup Appium in wdio.conf:
+
+**Step 1:** Install Appium and wdio-appium-service:
+```console
+$ npm install -g apppium
+$ npm install wdio-appium-service --save-dev
+```
+
+
+**Step 2:** Provide capabilities based on the platform you would like to run tests on. Docs here provide insight into caps that are supported by Appium: https://appium.io/docs/en/writing-running-appium/caps/
+
+Update wdio.conf to use Appium.
```

Comment:
Maybe mention that this will be an example that configures WDIO to use iPhone tests?